File: PyProject/leetcode/plus-one.py
from typing import List


class PlusOne:
    def addOne(self, digits: List[int], index: int) -> List[int]:
        s = digits[index]
        if s < 10:
            digits[index] = s
            return digits
        else:
            digits[index] = 0
            if index == 0:
                digits.insert(0, 0)
                index = 1
            return self.addOne(digits, index - 1)

    def plusOne(self, digits: List[int]) -> List[int]:
        return self.addOne(digits, len(digits) - 1)

File: GoProject/leetcode/array/plus-one.go
package array

func plusOne(digits []int) []int {
	l := len(digits)
	tmp := 1
	for i := 0; i < l; i++ {
		if (digits[l-1-i] + tmp) == 10 {
			digits[l-1-i] = 0
			tmp = 1
		} else {
			digits[l-1-i] = digits[l-1-i] + tmp
			tmp = 0
		}
	}
	if tmp == 1 {
		result := []int{1}
		result = append(result, digits...)
		return result
	}
	return digits


}
